### PR TITLE
chore: update/clarify changelog and commit instructions in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by ianbotsf**
## Issue \#

(none)

## Description of changes

Update and clarify the contributing guidelines to more explicitly detail:
* Expected format for changelog entries
* Optionality of subject and footer for commit messages

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by lucix-aws — APPROVED** [review]
shipit

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -120,67 +118,22 @@ The scope is optional but should be included when possible and refer to a module
...
 Breaking changes should also add an exclamation mark `!` after the type/scope (e.g. `refactor(rt)!: drop support for Android API < 20`)
 
-### Automated PR checks
-
-A number of automated workflows run when a PR is submitted. Generally speaking, each of these must pass before the PR is
-allowed to be merged. If your PR fails one of the checks, please attempt to address the problem and push a new commit to
-the PR. If you need help understanding or resolving a PR check failure, please reach out via a PR comment or a GitHub
-discussion. Please file a new issue if you believe there's a pre-existing bug in a PR check.
-
-#### Lint
```
Comment (lauzadis) [inline CONTRIBUTING.md:146]: I think this information about linting is helpful for new contributors -- personally I did not know about `ktlint` before joining.

I see we have the same section about linting in [aws-sdk-kotlin](https://github.com/awslabs/aws-sdk-kotlin/blob/main/CONTRIBUTING.md#lint), so was this removed mainly for deduplication?

**Review by lauzadis — APPROVED** [review]
Looks good, thanks for clearing up the optionality of the body and footer. I have one question but it does not block my approval

File: CONTRIBUTING.md (hunk 1 above)
Comment (ianbotsf) [reply CONTRIBUTING.md:146]: Whoops, this was unintentionally removed. It should absolutely still be here.

**Pull request merged**